src/app/submissions/submissions.component.html

```diff
@@ -40,8 +40,8 @@
       </ng-container>
       <ng-container matColumnDef="user">
         <mat-header-cell *matHeaderCellDef mat-sort-header i18n>Submitted by</mat-header-cell>
-        <mat-cell *matCellDef="let element">
-          <span *ngIf="element.submittedBy !== ''; else unknownLabel">{{element.submittedBy}}</span>
+        <mat-cell *matCellDef="let element" (click)="$event.stopPropagation()">
+          <span *ngIf="element.submittedBy !== ''; else unknownLabel" [routerLink]="name !== '' ? ['/users/profile/',element.submittedBy]:[]">{{element.submittedBy}}</span>
```

Comment:
You should use `element.user.name` (in a mat-table, element is the value of each element of the array/iterable supplied to the table) and check it does not equal `undefined`.

This code will always end up as "true" because for the value of `name` it will check the component, see there is no property `name` and the final comparison will be `undefined !== ''`.

On the `user` property of each submission, `name` will be a string if the user has created a profile or undefined if there is no profile.

Reply:
I've made the changes. When I ran it with ```user``` and ```name```, it seemed to have worked. In other words, I was only able to view the user profiles if the ```user !== ''```. Maybe this worked due to some loophole. Either way, I've used ```element.user.name``` this time and that works.